allennlp/commands/dry_run.py

```diff
@@ -0,0 +1,199 @@
+
+        return subparser
+
+
+def dry_run_from_args(args: argparse.Namespace):
+    """
+    Just converts from an ``argparse.Namespace`` object to params.
+    """
+    parameter_path = args.param_path
+    serialization_dir = args.serialization_dir
+    overrides = args.overrides
+
+    params = Params.from_file(parameter_path, overrides)
+
+    dry_run_from_params(params, serialization_dir)
+
+def dry_run_from_params(params: Params, serialization_dir: str) -> None:
+    prepare_environment(params)
+
+    vocab_params = params.pop("vocabulary", {})
+    vocab_dir = vocab_params.pop('directory_path', None)
+
+    if vocab_dir is not None:
+        logger.info("Found a directory_path parameter in your config. "
+                    "Also saving the vocab we create to that location.")
+        os.makedirs(vocab_dir, exist_ok=True)
+
+    os.makedirs(serialization_dir, exist_ok=True)
+
+    all_datasets = datasets_from_params(params)
+    datasets_for_vocab_creation = set(params.pop("datasets_for_vocab_creation", all_datasets))
+
+    for dataset in datasets_for_vocab_creation:
+        if dataset not in all_datasets:
+            raise ConfigurationError(f"invalid 'dataset_for_vocab_creation' {dataset}")
+
+    logger.info("Creating a vocabulary using %s data.", ", ".join(datasets_for_vocab_creation))
+
+    instances = [instance for key, dataset in all_datasets.items()
+                 for instance in dataset
+                 if key in datasets_for_vocab_creation]
+
+    vocabulary = verbosely_create_vocabulary(vocab_params, instances)
+
+    logger.info(f"writing the vocabulary to {serialization_dir}.")
+    vocabulary.save_to_files(os.path.join(serialization_dir, "vocabulary"))
+    if vocab_dir is not None and os.path.exists(vocab_dir) and os.listdir(vocab_dir) is not None:
+        logger.info(f"You passed a directory_path in your config which already exists and is non-empty. "
+                    f"Refusing to overwrite - we saved it to {serialization_dir} instead.")
+    elif vocab_dir is not None:
+        logger.info(f"You passed a directory_path in your config which was empty. Also "
+                    f"writing the vocabulary to {vocab_dir}.")
+        vocabulary.save_to_files(vocab_dir)
+
+
+def verbosely_create_vocabulary(vocab_params: Params, instances: List[Instance]) -> Vocabulary:
+    """
+    Given a parameter config specifying a vocabulary and a list of instances, this function
+    creates a Vocabulary from the instances. Additionally, it logs corpus statistics, prints
+    a random selection of instances and prints vocabulary sizes and namespaces.
+
+    Parameters
+    ----------
+    vocab_params : ``Params``, required
+        The parameters for the Vocabulary we create.
+    instances : ``List[Instance]``, required.
+        The instances to build the vocabulary from.
+
+    Returns
+    -------
+    The created Vocabulary.
+    """
+    min_count = vocab_params.pop("min_count", None)
```

Comment:
Why can't you just call `Vocabulary.from_params` here?  It's because you want to have access to `namespace_token_counts` when you're done, right?  If that's true, I'd make sure to note it in a comment here, to explain why there's a bunch of duplicated code.